lxd/api_cluster.go

```diff
@@ -164,12 +225,69 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
 	// Asynchronously join the cluster.
 	run := func(op *operation) error {
 		logger.Debug("Running cluster join operation")
-		// First request for this node to be added to the list of
-		// cluster nodes.
+
+		// If the user has provided a cluster password, setup the trust
+		// relationship by adding our own certificate to the cluster.
+		if req.ClusterPassword != "" {
+			err = cluster.SetupTrust(string(cert.PublicKey()), req.ClusterAddress,
+				string(req.ClusterCertificate), req.ClusterPassword)
+			if err != nil {
+				return errors.Wrap(err, "failed to setup cluster trust")
+			}
+		}
+
+		// Connect to the target cluster node.
 		client, err := lxd.ConnectLXD(fmt.Sprintf("https://%s", req.ClusterAddress), args)
 		if err != nil {
 			return err
 		}
+
+		// If node-specific configuration values are provided, it means
+		// the user wants to also initialize storage pools and
+		// networks.
+		if len(req.MemberConfig) > 0 {
+			// Connect to ourselves to initialize storage pools and
+			// networks using the API.
+			d, err := lxd.ConnectLXDUnix("", nil)
+			if err != nil {
+				return errors.Wrap(err, "failed to connect to local LXD")
+			}
+
+			err = clusterInitMember(d, client, req.MemberConfig)
+			if err != nil {
+				return errors.Wrap(err, "failed to initialize member")
```

Comment:
Errors must start with a capital (and never end with a period)

Reply:
Fixed.